File: apps/java/src/main/java/com/functions/HelloHttpFunction.java
package com.functions;

import com.microsoft.azure.functions.*;
import com.microsoft.azure.functions.annotation.*;
import java.util.Optional;

/**
 * HTTP hello endpoint — basic greeting with optional name parameter.
 */
public class HelloHttpFunction {

    @FunctionName("helloHttp")
    public HttpResponseMessage run(
            @HttpTrigger(
                name = "req",
                methods = {HttpMethod.GET},
                authLevel = AuthorizationLevel.ANONYMOUS,
                route = "hello/{name=world}")
            HttpRequestMessage<Optional<String>> request,
            @BindingName("name") String name,
            final ExecutionContext context) {

        String greeting = (name != null && !name.isEmpty()) ? name : "world";
        context.getLogger().info("Handled hello for " + greeting);

        return request.createResponseBuilder(HttpStatus.OK)
                .header("Content-Type", "application/json")
                .body("{\"message\":\"Hello, " + greeting + "\"}")
                .build();
    }
}

File: apps/dotnet/Functions/HelloHttpFunction.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Extensions.Logging;

namespace AzureFunctionsGuide.Functions;

/// <summary>
/// Hello HTTP handler — primary endpoint used across all tutorials.
/// Route: GET /api/hello/{name?}
/// </summary>
public class HelloHttpFunction
{
    private readonly ILogger<HelloHttpFunction> _logger;

    public HelloHttpFunction(ILogger<HelloHttpFunction> logger)
    {
        _logger = logger;
    }

    [Function("helloHttp")]
    public IActionResult Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "hello/{name?}")] HttpRequest req,
        string? name)
    {
        name ??= req.Query["name"].FirstOrDefault() ?? "world";
        _logger.LogInformation("Handled hello for {Name}", name);

        return new OkObjectResult(new { message = $"Hello, {name}" });
    }
}
